src/Codeception/Configuration.php

```diff
@@ -570,7 +570,22 @@ public static function isEmpty()
      */
     public static function append(array $config = [])
     {
-        return self::$config = self::mergeConfigs(self::$config, $config);
+        self::$config = self::mergeConfigs(self::$config, $config);
+        
+        if (isset(self::$config['paths']['log'])) {
```

Comment:
I was thinking about overriding by environment and it is not good idea because the results (HTML/XML reports) will be divided to separate folders by suites and I do not think it is desirable state. I think, user wants to set log path for run and not for suite. And for run can by path set by codeception.yml or --override parameter.